packages/medusa/src/services/event-bus.ts

```diff
@@ -278,147 +150,25 @@ export default class EventBusService {
   }
 
   async enqueuer_(): Promise<void> {
-    while (this.shouldEnqueuerRun) {
-      const listConfig = {
-        relations: [],
-        skip: 0,
-        take: 1000,
-      }
+    const listConfig = {
+      relations: [],
+      skip: 0,
+      take: 1000,
+    }
 
-      const stagedJobRepo = this.manager_.getCustomRepository(
-        this.stagedJobRepository_
-      )
-      const jobs = await stagedJobRepo.find(listConfig)
+    while (this.shouldEnqueuerRun) {
+      const jobs = await this.stagedJobService_.list(listConfig)
 
       await Promise.all(
-        jobs.map((job) => {
-          this.queue_
-            .add(
-              { eventName: job.event_name, data: job.data },
-              job.options ?? { removeOnComplete: true }
-            )
-            .then(async () => {
-              await stagedJobRepo.remove(job)
-            })
+        jobs.map(async (job) => {
+          return await this.eventBusModuleService_
+            .emit(job.event_name, job.data, job.options)
+            .then(async () => await this.stagedJobService_.remove(job))
         })
       )
 
+      // TODO: Refactor to a sleeper with exponential backoff and make it configurable
```

Comment:
thought: I don't think it would make sense to create an exponential backoff on this. The sleep time is the rate at which jobs are fetched from StagedJobs and placed into the EventBusModule. If this time increases exponentially, it will not take long before you have to wait a very long for the next batch of StagedJobs to arrive.

Reply:
If we were to add exponential backoff, we would need to "reset" the sleeper every time a staged job is added to the DB. That way, the backoff only kicks in in times where no events are being emitted in the system, and we avoid unnecessary polling. But I don't think this is super important, so I'll remove the todo and leave it for a follow-up iteration - if relevant.